src/sentry/metrics/base.py

```diff
@@ -78,3 +78,18 @@ def distribution(
         stacklevel: int = 0,
     ) -> None:
         raise NotImplementedError
+
+    def event(
+        self,
+        key: str,
+        value: str,
+        alert_type: str | None = None,
+        aggregation_key: str | None = None,
+        source_type_name: str | None = None,
+        date_happened: int | None = None,
+        priority: str | None = None,
+        instance: str | None = None,
+        tags: Tags | None = None,
+        stacklevel: int = 0,
+    ) -> None:
+        raise NotImplementedError
```

Comment:
This will backfire when we call `.event()` and the backend configuration uses a `MetricsBackend` implementation that doesn't support this method.

For example, we are running in production the `CompositeExperimentalBackend` so it will crash Sentry immediately if we deploy this PR and call `.event()`.

I think we might have two ways of proceeding:
1. We add `pass` in the parent method.
2. We implement the `event` method in each subclass with `pass` in the body.

Considering the complexity of the change I would opt for 1:
```suggestion
        pass
```

Reply:
Reverted to raising error now that the method is implemented for every derived class.
Lemme know if we should still `pass`.

Reply:
Please stick to NotImplementedError here. This is an abstract class (even if it is regrettably not defined as such), so it must not be instantiated directly. As a matter of fact all the other methods in this class are raising the same exception.

The right solution is to provde a valid implementation in all subclasses, like Niraj did.

Reply:
@nirajdewani  Have you checked if there is any subclass of this one in the `getsentry` repo?
If there is some there you will have to either add the method implementation first there and then make the change here, or actually make it temporarily `pass` here, then implement the method there and then revert this one to NotImplementedError

Reply:
Thanks, I checked; `InternalMetrics` in `src/sentry/utils/metrics.py` was the only subclass that was missed.
So should be okay to keep `raise NotImplementedError`.